src/common/sentry.ts

```diff
@@ -20,96 +20,68 @@
  */
 
 import { CaptureConsole, Dedupe, Offline } from "@sentry/integrations";
-import type { Event as SentryEvent } from "@sentry/types";
-import { sentryDsn, sentryDsnIsValid, isProduction } from "./vars";
+import { sentryDsn, isProduction } from "./vars";
 import { UserStore } from "./user-store";
 import logger from "../main/logger";
 
-// https://www.electronjs.org/docs/api/process#processtype-readonly
-const electronProcessType = ["browser", "renderer", "worker"] as const;
-
-export const integrations = [
-  new CaptureConsole({ levels: ["error"] }),
-  new Dedupe(),
-  new Offline()
-];
-
-const initialScope = (processType: typeof electronProcessType[number]) => {
-  return {
-    tags: {
-      // "translate" browser to 'main' as Lens developer more familiar with the term 'main'
-      "process": processType === "browser" ? "main" : "renderer"
-    }
-  };
-};
-
-export const environment = isProduction ? "production" : "development";
-
-const logInitError = (reason: string) => {
-  logger.warn(`⚠️  [SENTRY-INIT]: ${reason}, Sentry is not initialized.`);
-};
-
-export const beforeSend = (event: SentryEvent) => {
-  const allowErrorReporting = UserStore.getInstance().allowErrorReporting;
-
-  if (allowErrorReporting) return event;
-
-  logger.info(`🔒  [SENTRY-BEFORE-SEND-HOOK]: allowErrorReporting: ${allowErrorReporting}. Sentry event is caught but not sent to server.`);
-  logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: === START OF SENTRY EVENT ===");
-  logger.info(event);
-  logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: ===  END OF SENTRY EVENT  ===");
-
-  // if return null, the event won't be sent
-  // ref https://github.com/getsentry/sentry-javascript/issues/2039
-  return null;
-};
-
-export const SentryInit = () => {
-  if (!sentryDsnIsValid) {
-    logInitError("Sentry DSN is not valid");
-
-    return;
+/**
+ * This function bypasses webpack issues.
+ *
+ * See: https://docs.sentry.io/platforms/javascript/guides/electron/#webpack-configuration
+ */
+async function requireSentry() {
+  switch (process.type) {
+    case "browser":
+      return import("@sentry/electron/dist/main");
+    case "renderer":
+      return import("@sentry/electron/dist/renderer");
+    default:
+      throw new Error(`Unsupported process type ${process.type}`);
   }
+}
 
-  const processType = process.type as typeof electronProcessType[number];
-
-  let Sentry;
-
+/**
+ * Initialize Sentry for the current process so to send errors for debugging.
+ */
+export async function SentryInit(): Promise<void> {
   try {
-    // if in main process 
-    if (processType === "browser") {
-      Sentry = require("@sentry/electron/dist/main");
-    }
-
-    // if in renderer process
-    if (processType === "renderer") {
-      Sentry = require("@sentry/electron/dist/renderer");
+    const Sentry = await requireSentry();
+
+    try {
```

Comment:
This is a nested try because I prefer `const` over `let`. I really wish that JS had switch as expressions but they are only in the stage 2 for ecmascript proposals.